homeassistant/components/switch/broadlink.py

```diff
@@ -101,10 +103,30 @@ def _learn_command(call):
                                              "Did not received any signal",
                                              title='Broadlink switch')
 
+    @asyncio.coroutine
+    def _send_packet(call):
+        packets = call.data.get('packet', [])
+        for packet in packets:
+            for retry in range(DEFAULT_RETRY):
+                try:
+                    payload = b64decode(packet)
+                    yield from hass.loop.run_in_executor(
+                        None, broadlink_device.send_data, payload)
+                    break
+                except (socket.timeout, ValueError):
+                    try:
+                        yield from hass.loop.run_in_executor(
+                            None, broadlink_device.auth)
+                    except socket.timeout:
+                        if retry == DEFAULT_RETRY-1:
+                            _LOGGER.error("Failed to send packet to device.")
+
     if switch_type in RM_TYPES:
         broadlink_device = broadlink.rm((ip_addr, 80), mac_addr)
-        hass.services.register(DOMAIN, SERVICE_LEARN + '_' + ip_addr,
-                               _learn_command)
+        hass.services.register(DOMAIN, SERVICE_LEARN + '_' +
+                               ip_addr.replace('.', '_'), _learn_command)
```

Comment:
It shouldn´t be a breaking change. I think the learn_command service was intended to be triggered by the ui service developer tools. If you try to use the learn_command service in the current version with "." as delimiter HA throws the following error:

homeassistant.bootstrap: Invalid config for [script]: Service broadlink.learn_command_192.168.0.107 does not match format <domain>.<name> for dictionary value @ data['script']['debug']['sequence'][0]['service']. Got 'broadlink.learn_command_192.168.0.107'. (See /config/configuration.yaml, line 353). Please check the docs at https://home-assistant.io/components/script/

I think the problem is that only one "." is allowed in the service name to seperate domain from service name.

If this is the problem the change in the ip suffix is more a fix than a breaking change because it was never possible to use the service within a script.